OpenKh.Patcher/PatcherProcessor.cs

```diff
@@ -279,5 +284,79 @@ private static void PatchAreaDataScript(Context context, List<AssetFile> sources
+                        };
+                    string sourceText = File.ReadAllText(context.GetSourceModAssetPath(source.Name));
+                    if (source.Name.Contains("enc"))
+                    {
+                        byte[] data = System.Convert.FromBase64String(sourceText);
+                        sourceText = System.Text.ASCIIEncoding.ASCII.GetString(data);
+                    }
+                    switch (source.Type)
+                    {
+                        case "trsr":
+                            var TrsrList = Kh2.SystemData.Trsr.Read(stream).ToDictionary(x => x.Id, x => x);
+                            var moddedTrsr = deserializer.Deserialize<Dictionary<int, Kh2.SystemData.Trsr>>(sourceText);
+                            foreach (KeyValuePair<int, Kh2.SystemData.Trsr> treasure in moddedTrsr)
+                            {
+                                TrsrList[treasure.Value.Id].ItemId = treasure.Value.ItemId;
+                            }
+                                Kh2.SystemData.Trsr.Write(stream.SetPosition(0), TrsrList.Values);
+                            break;
+
+                        case "item":
+                            var ItemList = Kh2.SystemData.Item.Read(stream);
+                            var moddedItem = deserializer.Deserialize<Kh2.SystemData.Item>(sourceText);
+                            ItemList = moddedItem;
+                            ItemList.Write(stream.SetPosition(0));
+                            break;
+
+                        case "fmlv":
+                            var FormList = Kh2.Battle.Fmlv.Read(stream).ToDictionary(x => String.Concat(x.FormFm, x.FormLevel), x => x);
+                            var moddedForms = deserializer.Deserialize<Dictionary<string, Kh2.Battle.Fmlv.Level>>(sourceText);
+                            foreach (KeyValuePair<string, Kh2.Battle.Fmlv.Level> form in moddedForms)
+                            {
+                                FormList[form.Key].Ability = form.Value.Ability;
+                                FormList[form.Key].LevelGrowthAbility = form.Value.LevelGrowthAbility;
+                                FormList[form.Key].Exp = form.Value.Exp;
+                            }
+                            Kh2.Battle.Fmlv.Write(stream.SetPosition(0), FormList.Values);
+                            break;
+
+                        case "lvup":
+                            var LevelList = Kh2.Battle.Lvup.Read(stream);
+                            var moddedLevels = deserializer.Deserialize<Kh2.Battle.Lvup>(sourceText);
+                            int i = 0;
+                            foreach (Kh2.Battle.Lvup.PlayableCharacter character in moddedLevels.Characters)
+                            {
+                                LevelList.Characters[i].Levels = character.Levels;
```

Comment:
Isn't the patching too simplistic? How can I say I just want to patch Sora's level 60 but not anything else?

Reply:
This would be ideal, however Kh2.Battle.Lvup.PlayableCharacter has no identifier for what characterID this should belong to, and Kh2.Battle.Lvup.PlayableCharacter.Level has no identifier for what level# this should apply to.

moddedLevels could either deserialize into a custom type, or the above classes could be updated to have some internally identifying information. Thoughts?